sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -171,6 +176,7 @@ static int skl_hda_audio_probe(struct platform_device *pdev)
 	if (!mach)
 		return -EINVAL;
 
+	hda_soc_card.dev = &pdev->dev;
```

Comment:
is this not a different issue? or is this even required to move this  here?

Reply:
or for that matter required at all. devm_snd_soc_register_card() might actually handle this already?

Reply:
> is this not a different issue? or is this even required to move this here?

@plbossart This is a bit out-of-place, but I wanted to use dev_warn() in skl_hda_fill_card_info and easiest way to do this was to assign hda_soc_card.dev early.